android/src/main/kotlin/co/doneservices/callkeep/CallKeep.kt

```diff
@@ -383,6 +392,33 @@ class CallKeep(private val channel: MethodChannel, private var applicationContex
         result.success(null)
     }
 
+    private fun displayCustomIncomingCall(packageName: String, className: String, icon: String, extra: HashMap<String, String>) {
+        val notificationManager = applicationContext.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+        var launchIntent = Intent()
+        launchIntent.setClassName(packageName, "$packageName.$className")
+        launchIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK)
+
+        for ((key, value) in extra) {
+            launchIntent.putExtra(key, value)
+        }
+
+        val pendingIntent = PendingIntent.getActivity(applicationContext, 0, launchIntent, 0)
+        val builder = NotificationCompat.Builder(applicationContext, "my_channel_id")
+
+        builder.setSmallIcon(applicationContext.resources.getIdentifier(icon, "drawable", applicationContext.packageName))
+        builder.setFullScreenIntent(pendingIntent, true)
+        builder.setOngoing(true)
+        builder.setCategory(NotificationCompat.CATEGORY_CALL)
+        builder.setVisibility(NotificationCompat.VISIBILITY_PUBLIC)
+        builder.setPriority(NotificationCompat.PRIORITY_MAX)
+
+        val id = Random.nextInt(1000, 10000)
```

Comment:
We might need to pass through this id to the custom screen so that it can cancel the notification...